### PR TITLE
Add Docker 20.10.13 to 20.10.17

### PR DESCRIPTION
**Pull request by mitulshah-suse**
Add install docker scripts from 20.10.13 to 20.10.17
Related to https://github.com/rancher/rancher/issues/38390

### CODE REVIEW

**Review by snasovich — CHANGES_REQUESTED** [review]
The changes themselves LGTM, but can we add Ubuntu 22.04 to OSes tested for these newly added Docker versions? One of the reasons we're adding them is to address https://github.com/rancher/rancher/issues/38362 so having tests validate they are working is something we want to do.

File: scripts/test-docker
Comment (rayandas) [reply scripts/test-docker:18]: Right. 22.04 is not running for any of the version in Drone rn. @mitulshah-suse you have to add 22.04 in `pkg/20.10.13/config.sh` (I think we can skip adding 22.04 for all the versions below 20.10.13).

Comment (mitulshah-suse) [reply scripts/test-docker:19]: Have updated the tests to run 22.04 image for 20.10.13+
There was an issue in the test-docker script, running in the incorrect folder due to which it was not picking up the config.sh files from the pkg folders. Have fixed that issue as well.

Comment (mitulshah-suse) [reply scripts/test-docker:19]: [PASS] ubuntu:22.04-20.10.15 exit code: 0
[PASS] ubuntu:22.04-20.10.14 exit code: 0
[PASS] ubuntu:22.04-20.10.17 exit code: 0
[PASS] ubuntu:22.04-20.10.16 exit code: 0
[PASS] ubuntu:22.04-20.10.13 exit code: 0

Comment (superseb) [reply scripts/test-docker:19]: I think the `config.sh` issue is fixed in https://github.com/rancher/install-docker/pull/103 by @jakefhyde

Comment (mitulshah-suse) [reply scripts/test-docker:19]: It was simply a matter of updating [this](https://github.com/rancher/install-docker/blob/master/scripts/test-docker#L8) line to add `/../pkg` I checked the file history and seems like it was missed when the `test-docker` file was originally split from `test` script. 
Below is the original file.
https://github.com/rancher/install-docker/commit/d49abfb1aaffd855a358b61d3984923b7d214829#diff-d7ad7c7666dae3e86d4d3a850f7c4df6dcde86454e9995001ea017e44a3d35d5L10 
Anyways, if we are merging the fix in https://github.com/rancher/install-docker/pull/103 i will need to revert my changes done here for the fix.
So please let me know accordingly.

Comment (rayandas) [reply scripts/test-docker:19]: If https://github.com/rancher/install-docker/pull/103 get's merged, you can revert scripts/test-docker changes and just rebase with master.

Comment (jakefhyde) [reply scripts/test-docker:19]: #103 is now merged.

Comment (mitulshah-suse) [reply scripts/test-docker:19]: Ok, will update my PR accordingly...

**Review by superseb — CHANGES_REQUESTED** [review]
I guess this only needs a commits squash

**Comment by mitulshah-suse** [comment]
@jakefhyde @superseb 
Any reason for removing the config.sh files from 20.10.8-12?
We are getting errors in `make generate` scripts now. Please advise.
```
+ echo '==> [ERROR] DOCKER_GIT_COMMIT var should be defined at pkg/20.10.10/config.sh'
==> [ERROR] DOCKER_GIT_COMMIT var should be defined at pkg/20.10.10/config.sh
+ exit 1
FATA[0001] exit status 1                                
make: *** [Makefile:11: generate] Error 1
```

**Comment by jakefhyde** [comment]
@mitulshah-suse I removed them since we weren't using the overrides for the image name and tag anymore, we can add them back if needed.

**Comment by jakefhyde** [comment]
@mitulshah-suse I removed them since we weren't using the overrides for the image name and tag anymore, we can add them back if needed.

**Comment by mitulshah-suse** [comment]
> @mitulshah-suse I removed them since we weren't using the overrides for the image name and tag anymore, we can add them back if needed.

The `make generate` command is failing after removing the config files. Just wanted confirmation that they can be added back.

**Comment by jakefhyde** [comment]
> The `make generate` command is failing after removing the config files. Just wanted confirmation that they can be added back. Will add them again in this PR itself. Let me know in case of any concerns.

Yeah no worries about adding them back, although not sure if we need 16.04 anymore in the `config.sh` files.

**Review by jakefhyde — APPROVED** [review]

**Review by superseb — APPROVED** [review]

**Review by rayandas — APPROVED** [review]
LGTM

**Comment by rayandas** [comment]
Is it fine if we merge this PR right now? Or should we wait till 2.6.7 is out? @jakefhyde @superseb

**Comment by superseb** [comment]
We have a dev location that can be used to test that does not influence the production location of scripts (https://github.com/rancher/install-docker#sync-releases), but I can't merge either:

```
Merging can be performed automatically once the requested changes are addressed.
```

@snasovich is the only one who requested changes AFAICS

**Comment by rayandas** [comment]
Mitul addressed Sergey's changes. In drone, it's passing. 
```
[PASS] ubuntu:22.04-20.10.15 exit code: 0
[PASS] ubuntu:22.04-20.10.14 exit code: 0
[PASS] ubuntu:22.04-20.10.17 exit code: 0
[PASS] ubuntu:22.04-20.10.16 exit code: 0
[PASS] ubuntu:22.04-20.10.13 exit code: 0
```
If we dismiss Sergey's review, it'll be merged.

**Review by snasovich — APPROVED** [review]

**Comment by snasovich** [comment]
Sorry for the delay on my end, my review is no longer blocking the merge.

**Pull request merged**
